core/emitter.js

```diff
@@ -0,0 +1,34 @@
+// Gekko uses a custom event emitter within the GekkoStream (the plugins) to guarantee
+// the correct order of events that are triggered by eachother. Turns sync events from
+// LIFO into a FIFO stack based model.
+//
+// More details here: https://forum.gekko.wizb.it/thread-56579.html
+
+const util = require('util');
+const events = require('events');
+const NativeEventEmitter = events.EventEmitter;
+
+const GekkoEventEmitter = function() {
+  NativeEventEmitter.call(this);
+  this.defferedEvents = [];
+}
+
+util.inherits(GekkoEventEmitter, NativeEventEmitter);
```

Comment:
You could use ES6 `extends` here, it's reference in the [official doc](https://nodejs.org/docs/latest/api/util.html#util_util_inherits_constructor_superconstructor) and available since node `4.9.1` https://node.green/

Reply:
good point, will do!